### PR TITLE
Add example with roks

### PR DESCRIPTION
**Pull request by annumberhocker**

### CODE REVIEW

File: examples/roks_classic_with_cp4i/main.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
+provider "ibm" {
+  region = var.region
+}
+
+data "ibm_resource_group" "rg" {
+  name = var.resource_group
+}
+
+module "classic_openshift_single_zone_cluster" {
+  source = "github.com/terraform-ibm-modules/terraform-ibm-cluster.git//modules/classic-openshift-single-zone"
```
Comment (umarali-nagoor) [inline examples/roks_classic_with_cp4i/main.tf:10]: Source should be pointing to hashicorp registry level.

module "cluster" {
  source  = "terraform-ibm-modules/cluster/ibm"
  version = "1.5.0"
  # insert the 17 required variables here
}

File: examples/roks_classic_with_cp4i/variables.tf [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+variable "roks_version" {
+  description = "The OpenShift version that you want to set up in your cluster."
+  type        = string
+  default     = null
+}
+
+variable "storage_class" {
+  default     = "ibmc-file-gold-gid"
```
Comment (umarali-nagoor) [inline examples/roks_classic_with_cp4i/variables.tf:93]: Why storage class value is hard-coded

File: test/cloudpak_integration_test.go [hunk 3]
```diff
@@ -8,7 +8,7 @@ package test
 
 import (
 	"testing"
```
Comment (umarali-nagoor) [inline test/cloudpak_integration_test.go:10]: Please check, github action for test scheduler ... its failing 

Error: ./cloudpak_integration_test.go:46:2: undefined: fmt
Error: ./cloudpak_integration_test.go:51:2: undefined: fmt
Error: ./cloudpak_integration_test.go:56:2: undefined: fmt

File: examples/roks_classic_with_cp4i/variables.tf (hunk 2 above)
Comment (annumberhocker) [reply examples/roks_classic_with_cp4i/variables.tf:93]: For this example, ROKS classic is being invoked, so this is the required storage class that the CP4I module relies on.

File: examples/roks_classic_with_cp4i/main.tf (hunk 1 above)
Comment (annumberhocker) [reply examples/roks_classic_with_cp4i/main.tf]: Done and tested

File: test/cloudpak_integration_test.go (hunk 3 above)
Comment (annumberhocker) [reply test/cloudpak_integration_test.go:10]: I am not sure what to do here.  Basically, the variables passed into the sample being run need to correlate to a place where the owner of this repo has access to provision a cluster.  So, what values should I list here for `region`, `zone`, `vlan`?

```
func TestAccIBMCP4I(t *testing.T) {
	t.Parallel()

	// Construct the terraform options with default retryable errors to handle the most common retryable errors in
	// terraform testing.
	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
		// The path to where our Terraform code is located
		TerraformDir: "../examples/roks_classic_with_cp4i",

		// Variables to pass to our Terraform code using -var options
		Vars: map[string]interface{}{
			"region"		       = "ca-tor"
			"worker_zone"          = "tor01"
			"resource_group"       = "default"
			"workers_count"        = 4
			"worker_pool_flavor"   = "b3c.16x64"
			"public_vlan"          = ""
			"private_vlan"         = ""
			"force_delete_storage" = true
			"project_name"         = "cp4i"
			"environment"          = "test"
			"owner"                = "terratest"
			"roks_version"         = 4.7
			"entitled_registry_key" = os.Getenv(CP_ENTITLEMENT) //pragma: allowlist secret
			"entitled_registry_user_email" = os.Getenv(CP_ENTITLEMENT_EMAIL)
		},
	})
```

Also, you need to set CP_ENTITLEMENT and CP_EMAIL in the git secrets of whatever account is running these tests.  

```
      - name: Run Test
        working-directory: test
        run: go test -v ./...
        env:
          IC_API_KEY: ${{ secrets.ACCESS_KEY }}
          CP_ENTITLEMENT: ${{ secrets.CP_ENTITLEMENT }}
          CP_ENTITLEMENT_EMAIL: ${{ secrets.CP_ENTITLEMENT_EMAIL }}
```

**Comment by annumberhocker** [comment]
Finally, I do not know why the validate is failing.  It is producing this error:

image

**Pull request closed without merging**